### PR TITLE
CEMS-1620 Dan/add template

### PR DESCRIPTION
**Pull request by danmcadams**
Implement add template endpoint

### CODE REVIEW

**Comment by danmcadams** [comment]
The two failures are on purpose, setup for two future endpoints.

**Comment by tempcke** [comment]
@danmcadams we do not put failing tests on purpose in a PR ... 
you should have made the PR first, then did your setup tests in a new branch  ... test failures should never be 'expected' in a PR

File: api/ApiHelper.php [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 class ApiHelper {
   public static function getBaseURL(): string {
```
[the remark this answers is not in the document]
Comment (tempcke) [reply api/ApiHelper.php:8]: okay I took the time to download the repo and figure out what your using this for

the location header can be set 4 ways

- `scheme://host/path`
- `//host/path` (scheme used to access this endpoint will be used automatically)
- `/path` (scheme://host used to access this endpoint will be used automatically)
- `relative/path` (not recommended)

so `$_SERVER` is not needed ,... use the absolute path option `/path` without scheme or host

File: tests/Api/TestCase.php [hunk 2]
```diff
@@ -41,6 +44,9 @@ class TestCase extends \HomeCEU\Tests\TestCase {
 
   protected function setUp(): void {
     parent::setUp();
+    $_SERVER['HTTP_HOST'] = self::HOST;
```
Comment (tempcke) [inline tests/Api/TestCase.php:47]: see, i knew it... get this garbage out of here lol

File: api/ApiHelper.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (danmcadams) [reply api/ApiHelper.php:8]: https://tools.ietf.org/html/rfc2616#section-14.30

**Pull request merged**